### ISSUE TITLE
Project-stub шаблон блока

### ISSUE THREAD

**Issue by MikNikIlin** [bem/bem-forum-content-ru#455]
Не могу разобраться с подключением своих блоков на разные страницы. 

Например, создал я в папке desktop.blocks папку с неким блоком и файл стилей для него. Этот блок может встречаться в неизменном виде на некоторых страницах проекта. Как сделать так, чтобы можно было подключать этот блок в нужном месте, не описывая каждый раз структуру блока в page.bemjson.js для каждой страницы, где он встречается? 
В php я бы просто сделал include в нужном месте на нужный файл.

Спасибо за внимание.

**Comment by apsavin** [comment]
Судя вот по этому [ишью](https://github.com/bem/bem-core/pull/720), должен работать require

**Comment by Guria** [comment]
Для того чтобы любой блок попал в сборку необходимо чтобы было его упоминание в bemjson по которому осуществляется сборка или должна быть явно задана зависимость от этого блока в любом из блоков который заведомо попадает в сборку.
Скорее всего вы используете на всех своих страницах блок `page` из `bem-core` или у вас есть свой подобный аналог. Тогда есть 2 варианта:
 * создать папку блока `page` на общем уровне (desktop.blocks или common.blocks), положить в него файл page.deps.js:
```js
({
  shouldDeps: 'my-block'
})
```
В этом случае блок `my-block` будет подключаться всегда когда используется блок `page`.

 * создать папку `blocks` в папке вашего бандла (где лежит исходный `bemjson`) и разместить в ней папку page с файлом `page.deps.js` из предыдущего пункта.
В этом случае блок `my-block` будет подключаться когда используется блок `page`, но только в тех бандлах, в которых вы разместите подобную зависимость.

@apsavin твой ишью совсем о другом и не позволит подключить css блока.

PS. У меня стойкое чувство дежавю, как будто точно в такой формулировке этот вопрос уже звучал.

**Comment by Guria** [comment]
[Подробнее про зависимости](https://ru.bem.info/technology/deps/about/)
[Как устроены БЭМ-проекты](https://github.com/enb-bem/enb-bem-techs/blob/master/docs/bem-project.md)
[Сборка бандла](https://github.com/enb-bem/enb-bem-techs/blob/master/docs/build-bundle.md) по BEMDECL
[Сборка страницы](https://github.com/enb-bem/enb-bem-techs/blob/master/docs/build-page.md) по BEMJSON
@tadatuta не нашёл в документации где освещена возможность создании папки блоков только для текущего бандла

**Comment by apsavin** [comment]
@Guria А мне кажется, это ты не так понял топикстартера)

Человек спрашивает про аналог include, чтобы не описывать структуру bemjson каждый раз. А не про попадание css в сборку.

**Comment by apsavin** [comment]
@Guria Была вот такая тема: http://webcache.googleusercontent.com/search?q=cache:64ymgLsYAxEJ:clubs.ya.ru/bem/replies.xml%3Fitem_no%3D2508+&cd=10&hl=ru&ct=clnk&gl=ru&lr=lang_en|lang_ru

Как ее найти не в кэше гугла - вопрос)

**Comment by tadatuta** [comment]
@Guria 

> не нашёл в документации где освещена возможность создании папки блоков только для текущего бандла

Боюсь, что мы действительно нигде отдельно про это не писали :(
Исправимся.

**Comment by tadatuta** [comment]
@apsavin Да, с поиском по форуму беда, [делаем sitemap](https://github.com/bem/bem-forum/pull/103).
А так указанный пост здесь: https://ru.bem.info/forum/-287/

**Comment by Guria** [comment]
@apsavin возможно ты прав или истина на самомделе где-то посередине
